python/setup_cluster/launch_nomad_jobs.py

```diff
@@ -0,0 +1,167 @@
+        self.admin_password = self.security["admin"]["password"]
+        self.genai_key = self.api["genai_key"]
+        self.autoscaling_enabled = str(self.autoscaling["enabled"]).lower()
+        self.autoscaler_max_count = str(self.autoscaling["max_count"])
+
+        self.ssh_client_handler = SSHClientHandler(
+            self.node_ssh_username,
+            self.web_ingress_ssh_username,
+            self.web_ingress_public_ip,
+            web_ingress_private_ip=self.web_ingress_private_ip,
+            logger=logger,
+        )
+
+    def get_acl_token(self):
+        command = "grep 'Secret ID' /opt/neuraldb_enterprise/nomad_data/task_runner_token.txt | awk '{print $NF}'"
+
+        use_jump = self.nomad_server_private_ip != self.web_ingress_private_ip
+        ip = self.nomad_server_private_ip if use_jump else self.web_ingress_public_ip
+        return self.ssh_client_handler.execute_commands(
+            [command], ip if use_jump else self.web_ingress_public_ip, use_jump
+        )
+
+    def submit_nomad_job(self, nomad_ip, hcl_template, **kwargs):
+
+        def replace_placeholders(filepath, replacements):
+            with open(filepath, "r") as file:
+                content = file.read()
+
+            self.logger.info(f"Initial Content: {content}")
+            self.logger.info(f"Replacement Content: {kwargs}")
+            for key, value in replacements.items():
+                content = content.replace(f"{{{{ {key} }}}}", str(value))
+
+            temp_file = tempfile.NamedTemporaryFile(delete=False, mode="w")
+            temp_file_path = temp_file.name
+
+            self.logger.info(f"Final Content: {content}")
+            # Write the modified content to the temporary file
+            with open(temp_file_path, "w") as temp_file:
+                temp_file.write(content)
+
+            return temp_file_path
+
+        # Function to submit job to Nomad
+        def submit_to_nomad(hcl_file_path, nomad_endpoint, token):
+            headers = {"Content-Type": "application/json", "X-Nomad-Token": token}
+            hcl_to_json_url = f"{nomad_endpoint}v1/jobs/parse"
+            submit_job_url = f"{nomad_endpoint}v1/jobs"
+
+            with open(hcl_file_path, "r") as file:
+                hcl_content = file.read()
+
+            # Convert HCL to JSON using the Nomad API
+            response = requests.post(
+                hcl_to_json_url,
+                headers=headers,
+                json={"JobHCL": hcl_content, "Canonicalize": True},
+            )
+            job_json = response.json()
+
+            # Submit the job JSON to Nomad
+            final_response = requests.post(
+                submit_job_url, headers=headers, json={"Job": job_json}
+            )
+            self.logger.info(final_response.text)  # Log the response for debugging
+
+            # Cleanup the temporary file
+            os.remove(hcl_file_path)
+
+        nomad_endpoint = f"http://{nomad_ip}:4646/"
```

Comment:
Probably can do this in a different PR, but just commenting here as well. We should move to SSHing into a node and then launching the job using localhost:4646, instead of using the public ip:4646. The reason being some companies block all ports except http/https/ssh, so they won't be able to access port 4646. Intuit is a specific example where they had to change the bash script to first ssh into the node and then launch the job using localhost:4646.

Reply:
Added a TODO.